Pennsieve/pennsieve-discover-app
# fix(Datasets): Fix race condition with embargoed dataset

# Description
The purpose of this PR is to fix a race condition with embargoed dataset. This is done by using a Nuxt plugin for Cookies instead of `js-cookie`. `js-cookie` was having a hard time reading the cookies during the `asyncData` phase of data fetching, which most pages rely on.

https://www.npmjs.com/package/cookie-universal-nuxt

## Clickup Ticket

[t9bjf0](https://app.clickup.com/t/t9bjf0)


## Type of change



# How Has This Been Tested?

- Login to the app
- Visit [Cameron's New Dataset for publishing](http://localhost:3000/datasets/2189) and request access if you need it. Please Slack me when requested and I can grant access.
  - If you already have access to an embargoed dataset, please use that dataset
- Once given access, the "Request dataset" button should be "Get Dataset"
- Refresh the page to direct load the dataset
- The "Request dataset" button should be "Get Dataset"

# Checklist:

- [x] I have pushed the final commit message using the [changelog format](https://github.com/Blackfynn/blackfynn-app/wiki/CHANGELOG)
